docs: update changelog for 0.17.0 (#915)
scikit-build/scikit-build: CHANGES.rst

We are hard at work on the next generation of scikit-build `scikit-build-core
<https://github.com/scikit-build/scikit-build-core>`_, which will eventually
replace the backend here. We are also continuing to fix bugs, make improvements,
and backport changes here.

.. START-BRIEF-CHANGELOG

Scikit-build 0.17.0
===================

A lot of bug fixes are present in this release, focusing on Windows, PyPy, and
cross compiling. We've also improved the compatibility with default setuptools
behaviors a little, and enabled some things that were previously unavailable,
like overriding the build type via the cmake argument environment variables.
We've expanded our CI matrix to include Windows and macOS PyPy and some Fortran
tests on Linux. This release requires Python 3.7+.

Bug fixes
---------

* Match setuptools behavior for ``include_package_data`` default. by :user:`vyasr` in :pr:`873`
* Misc. fixes for F2PY and PythonExtensions modules by :user:`benbovy` in :pr:`495`
* Provide more useful error if user provides ``CMAKE_INSTALL_PREFIX`` by :user:`vyasr` in :pr:`872`
* Stop assuming that ``.pyx`` files are in the same directory as ``CMakeLists.txt`` by :user:`vyasr` in :pr:`871`

> * Allow build type overriding by :pr:`902`

Reviewer: ```suggestion
* Allow build type overriding by user:`henryiii` in :pr:`902`
```

> * Detect PyPy library correctly on Windows (#701) in :pr:`904`

Reviewer: ```suggestion
* Detect PyPy library correctly on Windows by user:`gershnik` in :pr:`904`
```

Author: Thanks, not sure what happened here.

> * Include library for FindPython for better Windows cross-compiles in :pr:`913`

Reviewer: ```suggestion
* Include library for FindPython for better Windows cross-compiles by user:`henryiii` in :pr:`913`. Thanks to user:`maxbachmann` for testing.
```

> * Use f2py's ``get_include`` if present by in :pr:`877`

Reviewer: ```suggestion
* Use f2py's ``get_include`` if present by user:`henryiii` in :pr:`877`
```

Author: I've generally avoided adding core maintainers in the changelog, and only listed external contributions. (For historical reasons, that's how it was done before here - in scikit-build-core, I just take the default output from GH, which includes the users).

Reviewer: Thanks for the review. Agreed, let's continue by not mentioning core maintainers.

Background: Since I noticed the wording `by in` for some of the entry, I was not sure of the original intent.

Author: Yes, I start with the GitHub generated release notes even here, so I’m manually filtering core maintainer mentions - so I missed deleting “by”.

> * Fix support for cross-compilation exception using ``targetLinkLibrariesWithDynamicLookup`` by :user:`erykoff` in :pr:`901`

Reviewer: ```suggestion
* Fix support for cross-compilation exception using ``targetLinkLibrariesWithDynamicLookup`` by :user:`erykoff` and :user:`jcfr` in :pr:`901`
```

Author: Same thing, I've generally avoided listing us. If you want to start (I don't care either way), they will all need updates.

Testing
-------

* Add hello fortran sample package + tests by :user:`benbovy` in :pr:`493`

> * Add sdist check & fix by in :pr:`906`
> * Fix some setuptools types by in :pr:`888`
> * Add PyPy Win & macOS to the CI in :pr:`907`

Reviewer: ```suggestion
* Add sdist check & fix by user:`henryiii` in :pr:`906`
* Fix some setuptools types by user:`henryiii` in :pr:`888`
* Add PyPy Win & macOS to the CI by user:`henryiii` in :pr:`907`
```

Miscellaneous
-------------

> * Drop Python 3.6 in :pr:`862`
> * Move building backend to hatchling in :pr:`870`
> * Avoid mutating function input parameters in :pr:`899`
> * Use _compat/typing name in :pr:`869`

Reviewer: ```suggestion
* Drop Python 3.6 by user:`henryiii` in :pr:`862`
* Move building backend to hatchling by user:`henryiii` in :pr:`870`
* Avoid mutating function input parameters by user:`henryiii` in :pr:`899`
* Use _compat/typing name by user:`henryiii` in :pr:`869`
```
